metapool/calculate_coefficients.py

```diff
@@ -0,0 +1,147 @@
+        # ax.scatter(group["CPMlog"], group["Dilution2"])
+
+        # TODO FIXME HACK: Unclear what R^2 value is being printed by ggplot
+        #  but it definitely isn't a pearson computed per group.  *shrug*
+        no_nan = group.dropna()
+        r, p = scipy.stats.pearsonr(no_nan[xcol], no_nan[ycol])
+        sns.regplot(x=xcol, y=ycol, data=group, ax=ax)
+        ax.text(textpos[0], textpos[1], 'r={:.2f}\np={:.2g}'.format(r, p),
+                transform=ax.transAxes)
+        if x > 0:
+            ax.set_ylabel("")
+        if y < ymax - 1:
+            ax.set_xlabel("")
+
+        ax.set_title("\n".join(name))
+        x += 1
+        if x == xmax:
+            x = 0
+            y += 1
+        if y == ymax:
+            plt.show()
+            x = 0
+            y = 0
+
+    if x != 0 or y != 0:
+        plt.show()
+
+
+def calculate_coefficients(table_synthetic_hits, metadata_pools, dilutions,
+                           plot_fit=False):
+    # Strip leading X in sample names which are automatically removed by R
+    # table reading.
+    metadata_pools["sample_name_r"] = metadata_pools["sample_name_r"].str[1:]
+
+    # Parsing the files
+    # Calculate the total number of reads aligned
+    # to the plasmid sequences for each sample
+    table_synthetic_hit_totals = table_synthetic_hits.sum()
+    table_synthetic_hit_totals = pd.DataFrame(
+        data={"sample_name_r": table_synthetic_hit_totals.index.to_list(),
+              "table_synthetic_hit_totals":
+                  table_synthetic_hit_totals.values.tolist()},
+        index=None)
+
+    # Convert feature-table to long-form and merge with total number of hits to
+    # plasmid sequences
+    table_synthetic_hits_long_with_totals_beta = pd.melt(
+        table_synthetic_hits.reset_index(), id_vars="OTUID")
+    table_synthetic_hits_long_with_totals_beta.columns = ["plasmid_id",
+                                                          "sample_name_r",
+                                                          "count"]
+    table_synthetic_hits_long_with_totals = \
+        table_synthetic_hits_long_with_totals_beta.merge(
+            table_synthetic_hit_totals, on="sample_name_r")
+
+    # Merge updated feature-table with sample-pool information
+    table_synthetic_hits_long_with_totals_and_pools = \
+        table_synthetic_hits_long_with_totals.merge(
+            metadata_pools, on="sample_name_r")
+
+    # Aggregate across lanes (not needed)
+    table_synthetic_hits_long_with_totals_and_pools_all_lanes = \
+        table_synthetic_hits_long_with_totals_and_pools
+
+    # Calculate counts per million
+    table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+        "counts_per_million"] = (
+            table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+                "count"] /
+            table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+                "read_count_total"]
+        ) * 1000000
+
+    # Calculate percentages
```

Comment:
should these use the rescale_counts._to_column_percentage? The same question applies to the lines/section below.

Reply:
I'll do you one better and just delete it.  It is copied from the R code which has already converted to a long form table and thus the to_column_percentage function is not applicable.  It is called a percentage, but divides by 100 rather than multiplying by 100, so is off by a factor of 10000.  It is also unused.